app/Models/Company.php

```diff
@@ -128,9 +128,12 @@ public static function isCurrentUserHasAccess($companyable)
         } elseif (!static::isFullMultipleCompanySupportEnabled()) {
             return true;
         } else {
-            $current_user_company_id = Auth::user()->company_id;
-            $companyable_company_id = $companyable->company_id;
-            return ($current_user_company_id == null || $current_user_company_id == $companyable_company_id || Auth::user()->isSuperUser());
+            if (Auth::user()) {
```

Comment:
I can't put my finger on why this makes me a little nervous. But I guess it's okay. I just worry that if you could somehow force this method to run when you're *not* logged-in to Snipe-IT, that you might somehow be able to circumvent access? I'm not sure if I'm being weird here though. And I can't really think of any other way to do it. Maybe if there were a way to say 'We are superuser' when we're running via CLI? Not sure.

Anyways, just a slight concern, and since I can't really articulate it very well I think I'm okay with it like this.